app/models/microservices_engine/connection.rb

```diff
@@ -24,34 +16,22 @@ def get(reqs, attrs)
       # the results.
 
       # Example use:
-      # (connection object for FieldTrip).data({active: true, bus: false}, {:id, :cost_estimate})
-      # => queries endpoint
-      # => endpoint finds all FieldTrip objects that are active and have the bus attribute being false
-      # => it then collects the `attrs` and return
-    end
+      # (connection object for FieldTrip).get(:trip, [123223, public_trip_stops], {active_only: true})
+      # => queries endpoint: uri/123223/public_trip_stops
+      # => endpoint finds all FieldTrip objects that are active (param flag)
+      # => returns the response if the request was a success
 
-    def self.get(options, reqs, attrs = [])
-      # VALID USES
-        # All possible fields present
-        # MicroserviceEngine.get({model: :public_trip_stops, through: :trip}, {id: 130732}, [:active, :id])
+      # Assumption: url is followed by a `/`
+      uri = URI.parse(url + path.join('/'))
+      uri.query = URI.encode_www_form(params)
 
-        # All optional fields dropped
-        # MicroserviceEngine.get({model: :trip}, {active: true})
-
-      # INVALID USES
-        # No model specified
-        # MicroserviceEngine.get({}, {active: true})
-
-        # Requirements skipped
-        # MicroserviceEngine.get({model: :trip}, [:id])
-
-      model = options.fetch(:model) rescue return {}
-      through = options.fetch(:through, nil)
-
-      conn = Connection.where(object: through || model)
+      res = Net::HTTP.get_response(uri)
+      return res.body if res.is_a?(Net::HTTPSuccess)
+    end
 
-      attrs = attrs + [model] unless through.nil?
-      conn ? conn.get(reqs, attributes) : {}
+    def self.get(resource, path, params = {})
+      conn = Connection.where(object: resource.to_s) # Does :abc match "abc"?
```

Comment:
It's a work in progress :upside_down_face:. It's near impossible to do TDD when the entire concept of the software changes like every third day.